### ISSUE TITLE
file uploading first reqest posting will shadow the post parameters.

### ISSUE THREAD

**Issue by calidion** [balderdashy/sails#2889]
I have found a bug not knowing if anyone ever reported.
I need to post image files and parameter updates to the server. There are two possible ways to send the data in sequences.
1. image last

```javascript
 for (var key in data) {
   req.field(key, data[key]);
 }
 req
   .attach('image', __dirname + '/assets/images/merchandise.jpg');
```

2.image first
```javascript
 req
   .attach('image', __dirname + '/assets/images/merchandise.jpg');
 for (var key in data) {
   req.field(key, data[key]);
 }
```

the test result differ due to the sequence difference.
method 1 will pass, and method 2 will failed due to the parameters can not be parsed by the request object .

**Comment by tjwebb** [comment]
> method 1 will pass, and method 2 will failed

Post the error message.

**Comment by calidion** [comment]
@tjwebb 
there are no error messages, but keys are missing from req.

**Comment by tjwebb** [comment]
Can you post a link to your code? You're not giving me a lot to go on here

**Comment by sailsbot** [comment]
Thanks for posting, @calidion.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
